figma-shop/src/app/pages/products/products.component.ts

```diff
@@ -10,18 +11,36 @@ import { ProductServiceService } from '../../shared/services/product-service.ser
 })
 
 export class ProductsComponent implements OnInit {
-  public allProducts: Array<ProductInterface>;
-  public productsViewed: Array<ProductInterface> = [];
+  public allProducts: ProductInterface[];
   public page = 1;
+  public itemsPerPage = 5;
+  public sortConditions = ['Default', 'price Small - Large', 'price Large - Small', 'rating Small - Large', 'rating Large - Small'];
+  public sortValue = 'Default';
 
   constructor(
-    private productService: ProductServiceService,
-    private actRoute: ActivatedRoute) { }
+    private productService: ProductService,
+    private actRoute: ActivatedRoute,
+    private filterService: FiltersService) { }
 
   ngOnInit(): void {
     this.actRoute.data.subscribe(data => {
       this.allProducts = data.products;
     });
```

Comment:
How about unsubscribe?
also, separete it

Reply:
OK